cdap-ui/graphql/resolvers/programRecordResolvers.js

```diff
@@ -0,0 +1,43 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+const programsResolver = {
+  ApplicationDetail: {
+    programs: async (parent, args, context, info) => {
+      return await (new Promise((resolve, reject) => {
```

Comment:
Do we have to wrap this in a `Promise`? Can't it be just synchronous? We getting programs from parent object and filtering then by type.

Reply:
Is it unnecessary to have it with a Promise? Is the logic too simple?

Reply:
Promise helps when co-ordinating with asynchronous tasks. For instance to fetch something from backend. We don't know when it will return back. So we return a promise and when backend returns back we resolve the promise. In this case it seems like we are doing a synchronous operation.